sklearn/model_selection/_split.py

```diff
@@ -862,6 +864,10 @@ def _iter_test_masks(self, X, y, groups):
             raise ValueError("The groups parameter should not be None")
         groups = np.array(groups, copy=True)
         unique_groups = np.unique(groups)
+        if self.n_groups > len(unique_groups):
+            raise ValueError("Cannot have n_groups=%d greater than the number "
+                             "of unique groups: %d."
```

Comment:
I think it's okay as `n_groups` is a parameter and unique groups is the information we are trying to include in the error message...